src/box/sql/mem.c

```diff
@@ -2095,7 +2095,7 @@ mem_add_num(const struct Mem *left, const struct Mem *right, struct Mem *result)
 		diag_set(ClientError, ER_SQL_EXECUTE, "integer is overflowed");
 		return -1;
 	}
-	mem_set_int(result, res, is_neg);
+	mem_set_int_with_sign(result, res, is_neg);
```

Comment:
What's the point in carrying the sign around? Why not just check if `res` is negative in `mem_set_int`?

IOW can you get rid of `mp_set_int_with_sign` and `mp_set_nint` and always use `mp_set_int` instead?

Reply:
Yeah, like Mergen said, we use `int64_t` as a container both for signed and unsigned integer. It might store `UINT64_MAX`, for example, but then we have to see that `is_neg == false`.